pkg/chains/formats/slsa/internal/material/material.go

```diff
@@ -60,40 +63,45 @@ func AddImageIDToMaterials(imageID string, mats *[]common.ProvenanceMaterial) er
 	m := common.ProvenanceMaterial{
 		Digest: common.DigestSet{},
 	}
-	uriDigest := strings.Split(imageID, uriSeparator)
-	if len(uriDigest) == 2 {
-		digest := strings.Split(uriDigest[1], digestSeparator)
-		if len(digest) == 2 {
-			// no point in partially populating the material
-			// do it if both conditions are valid.
-			uri := strings.TrimPrefix(uriDigest[0], "docker-pullable://")
-			m.URI = artifacts.OCIScheme + uri
-			m.Digest[digest[0]] = digest[1]
-			*mats = append(*mats, m)
-		} else {
-			return fmt.Errorf("expected imageID %s to be separable by @ and :", imageID)
-		}
-	} else {
-		return fmt.Errorf("expected imageID %s to be separable by @", imageID)
+	uriDigest, err := extractUriDigestFromImageID(imageID)
+	if err != nil {
+		return err
 	}
+	m.URI = uriDigest[URI]
+	m.Digest[uriDigest[DigestAlgorithm]] = uriDigest[DigestValue]
+	*mats = append(*mats, m)
 	return nil
 }
 
+// extractUriDigestFromImageID extracts uri and digest from an imageID with format <uri>@sha256:<digest>
+func extractUriDigestFromImageID(imageID string) (map[string]string, error) {
+	uriDigest := strings.Split(imageID, uriSeparator)
+	if len(uriDigest) != 2 {
+		return map[string]string{}, fmt.Errorf("expected imageID %s to be separable by @", imageID)
+	}
+	digest := strings.Split(uriDigest[1], digestSeparator)
+	if len(digest) != 2 {
+		return map[string]string{}, fmt.Errorf("expected imageID %s to be separable by @ and :", imageID)
+	}
+	uri := strings.TrimPrefix(uriDigest[0], "docker-pullable://")
+	return map[string]string{URI: artifacts.OCIScheme + uri, DigestAlgorithm: digest[0], DigestValue: digest[1]}, nil
+}
+
```

Comment:
One imageID can only produce one uri, one digestAlgo and one digest. Why do we need a map here? Would it be a good idea to return 3 named values instead?